cas/cas.go

```diff
@@ -264,6 +307,17 @@ func (ds Store) WriteACI(r io.Reader, latest bool) (string, error) {
 // Users of treestore should call ds.RenderTreeStore before using it to ensure
 // that the treestore is completely rendered.
 func (ds Store) RenderTreeStore(key string, rebuild bool) error {
+	// this lock references the treestore dir for the specified key. This
+	// is different from a lock on an image key as internally
+	// treestore.Write calls the acirenderer functions that use GetACI and
+	// GetImageManifest which are taking the image lock.
+	treestoreLockKey := key + "-treestore"
```

Comment:
> In 8bbbde4 we document the semantics of taking the lock. We need to ensure we document those semantics on the function that generates the lock name so a user of the lock knows what she can/cannot do while holding the lock.

Would you prefer to document the locking semantics in every function or at the start of cas.go?